frontend/lobby/team.tsx

```diff
@@ -56,30 +62,61 @@ class _TeamComponent extends React.PureComponent<Props, State> {
-                {this.props.otherTeamMembers.map(member =>
-                    <li key={member.id}>{member.name} <OnlineIndicator when={member.online}/> <AdminIndicator when={member.isAdmin}/> </li>
-                )}
-            </ul>
+        const { teamView } = this.state;
+        return <div className="teams">
+            <nav className="tabs">
+                <button onClick={this.switchTab} value="team" className={teamView ? 'active' : ''}>Team</button>
+                <button onClick={this.switchTab} className={teamView ? '' : 'active'}>Leaderboards</button>
+            </nav>
+            { teamView &&
+                <div>
+                    <h1>{this.props.teamName}</h1>
+                    <div className="saltines-count">
+                        <h3>Saltines</h3>
+                        <p><img height="20" width="20" src={saltine} alt="Saltine" /><span>{this.props.totalPoints}</span>(current balance)</p>
+                        <p><img height="20" width="20" src={saltine} alt="Saltine" /><span>{this.props.totalPointsAllTime}</span>(earned all-time)</p>
+                    </div>
+                    <div className="rankings-snapshot">
+                        <h3>Rank</h3>
+                        <button onClick={this.switchTab}><sup>#</sup>2<span>in Vancouver</span></button>
+                        <button onClick={this.switchTab}><sup>#</sup>78<span>in Canada</span></button>
+                    </div>
+                    <h3>Your team</h3>
+                    <ul className="team">
+                        <li>{this.props.isTeamAdmin && <button onClick={this.switchTeamAdmin}>X</button>}{this.props.myName} (That's you!) <OnlineIndicator when={this.props.isOnline}/> <AdminIndicator when={this.props.isTeamAdmin}/> </li>
+                        {this.props.otherTeamMembers.map(member =>
+                            <li key={member.id}>{member.name} <OnlineIndicator when={member.online}/> <AdminIndicator when={member.isAdmin}/>{this.props.isTeamAdmin && <button value={member.name} onClick={this.removeTeamMember}>X</button>} </li>
+                        )}
+                        {this.props.otherTeamMembers.length < 4 && <li><button onClick={this.addTeamMemberPrompt}>Add Team Member</button></li>}
+                    </ul>
+                </div>
+            }
+            { !teamView &&
+                <div>
+                    <p>Here be the leaderboards component.</p>
+                </div>
+            }
+            
         </div>;
     }
+
+    @bind private switchTab(event: React.MouseEvent<HTMLButtonElement>) {
+        const tab = event.currentTarget.value
+        tab ? this.setState({ teamView: true }) : this.setState({ teamView: false })
+    }
+
+    @bind private addTeamMemberPrompt(event: React.MouseEvent<HTMLButtonElement>) {
+        confirm('To add someone to your team, give them your team code: ' + this.props.teamCode + '. Your team can have a maximum of 5 people.')
```

Comment:
Can we get rid of the "Add Team Member" button and just put this text there in its place, below the list of users? (When the team isn't full.) I don't see why it should be a button.

Reply:
Yeah good point. Can `this.props.otherTeamMembers.length` be used reliably? Is that a measure of all players excluding yourself? Or is it all non-admin players excluding yourself?

Reply:
Yes, `this.props.otherTeamMembers.length` is a reliable count of all players excluding yourself (it includes admins)